### PR TITLE
Flask Service

### PR DESCRIPTION
**Pull request by MrHardcode**
~~Experimental branch with one hardcoded example of how to make the flask server work~~
Working on a proper setup of the flask server on this branch now

 * [x] Add try-catch in case we try to plot data, when all data has been filtered away or there simply was no available data to begin with
 * [x] Add support for more filters in the endpoint
 * [x] Improve readability in code
 * [x] Decide how to handle the amount of tweets scraped

### CODE REVIEW

**Comment by MrHardcode** [comment]
I've made some mistakes so don't merge anything

File: modules/presentation.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def pie_chart(df, title, save=None):
     plt.ylabel("")
 
     if save:
-        save_plot(plt, save)
+        save_plot(plt, title)
```
Comment (MrHardcode) [inline modules/presentation.py]: I misunderstood the use of the save parameter. Thought it was meant to be used as a boolean. Should be changed back

**Comment by MrHardcode** [comment]
Plotting still seems to be bugged - check out [this](https://github.com/Hold-Krykke/PythonExam/blob/174aae59855b7acd1c6d94a0d0652e1806cd6bbe/modules/plots/%40JoeBiden.png)

Plot was created with the following request data:
```JSON
{
    "hashtags": [
        "trump",
        "biden"
    ],
    "start_date": "2020-5-3",
    "end_date": "2020-5-12",
    "plot_type": "line",
    "search_for": {"mentions": "@JoeBiden"}
}
```

**Comment by MrHardcode** [comment]
Should the amount of tweets scraped be an option or should we just always scrape 500 as standard? (would get more older tweets). I vote for the hardcoded option of course

**Comment by Castau** [comment]
Either could be an option. For the purpose of the project it's fine to limit it to a specific amount :)

**Comment by MalteMagnussen** [comment]
> Should the amount of tweets scraped be an option or should we just always scrape 500 as standard? (would get more older tweets). I vote for the hardcoded option of course

Depends how long it takes? 
If it takes barely any time to scrape, lets say, 2000 tweets, compared to 500, then why not let user do that? (Hardcoded option) 
If its easy to up the amount of tweets scraped, then it should also be relatively straightforward to let user pass a body parameter to take more. 
Default 500, if no parameter set by user in JSON body?

**Comment by MrHardcode** [comment]
Okay let's make a default at 300 (takes ca. 10 seconds) and let the user change the amount

**Comment by MrHardcode** [comment]
Okay now you have the option of controlling the web scraper 
image

File: modules/Sentiment_Analysis.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 from nltk import classify
 from nltk import NaiveBayesClassifier
 from nltk.tokenize import word_tokenize
-from modules.Preprocessing import remove_noise
+from Preprocessing import remove_noise
```
Comment (MrHardcode) [inline modules/Sentiment_Analysis.py:6]: Imports only work for me, when written like this

File: modules/flask_service.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+import presentation
+import web_scraper
+import Preprocessing
+import Sentiment_Analysis
+from datetime import datetime, date 
+
+app = Flask(__name__)
+
+@app.route('/api/sentiment', methods=['POST'])
+def get_burglaries():
```
Comment (MalteMagnussen) [inline modules/flask_service.py]: burglaries?

File: modules/flask_service.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+    plot_type = get_plot_type(request)
+
+    start_date, end_date = get_dates(request)
+
+    # Creating plot
+    try:
+        prepare_data_and_create_plot(hashtags, tweet_amount, fresh_search, file_name, start_date, end_date, plot_type, search_for, rm_sentiment)
+    except:
+        return jsonify({"Message": "All data has been filtered away - unable to create plot"})
```
Comment (MalteMagnussen) [inline modules/flask_service.py:37]: Nice

File: modules/presentation.py [hunk 5]
```diff
@@ -77,9 +79,10 @@ def get_by_key_value(tweets, key, value):
     Returns: \n
         Filtered List
     """
-    if key not in ["hashtags", "people", "urls"]:
+    if key not in ["hashtags", "mentions", "tweet_urls"]:
```
Comment (MalteMagnussen) [inline modules/presentation.py:82]: thanks, sorry this wasnt fixed by me

**Review by MalteMagnussen — APPROVED** [review]

**Review by Runi-VN — APPROVED** [review]

**Pull request merged**